pkgs/tools/networking/persepolis/default.nix

```diff
@@ -0,0 +1,69 @@
+{ stdenv, buildPythonApplication, fetchFromGitHub, makeDesktopItem
+, aria
+, libnotify
+, pulseaudio
+, psutil
+, pyqt5
+, requests
+, setproctitle
+, sound-theme-freedesktop
+, youtube-dl
+}:
+
+buildPythonApplication rec {
+  pname = "persepolis";
+  version = "3.1.0";
+
+  src = fetchFromGitHub {
+    owner = "persepolisdm";
+    repo = "persepolis";
+    rev = "${version}";
+    sha256 = "0xngk8wgj5k27mh3bcrf2wwzqr8a3g0d4pc5i5vcavnnaj03j44m";
+  };
+
+  desktopItem = makeDesktopItem {
+    name = "persepolis";
+    exec = "persepolis";
+    comment = "Download Manager";
+    icon = "persepolis";
+    desktopName = "persepolis";
+    genericName = "Download Manager";
+    categories = "Qt;Network;";
+  };
+
+  preBuild=''
+    substituteInPlace setup.py --replace "answer = input(" "answer = 'y'#"
+  '';
+
+  postPatch = ''
+    sed -i 's|/usr/share/sounds/freedesktop/stereo/|${sound-theme-freedesktop}/share/sounds/freedesktop/stereo/|' setup.py
+    sed -i "s|'persepolis = persepolis.__main__'|'persepolis = persepolis.scripts.persepolis:main'|" setup.py
+  '';
+
+  postInstall = ''
+     mkdir -p $out/share/applications
+     cp ${desktopItem}/share/applications/* $out/share/applications
+  '';
+
+  # see: https://github.com/persepolisdm/persepolis/blob/3.1.0/setup.py#L130
+  doCheck = false;
+
+  propagatedBuildInputs = [
+    aria
+    libnotify
```

Comment:
After reading [setup.py](https://github.com/persepolisdm/persepolis/blob/master/setup.py) I'm wondering if `aria` and possibly `libnotify` don't belong in propagatedBuildInputs, but instead if the program should use `makeWrapper` to include `aria` and `libnotify` in the PATH?

Reply:
Something like this:
https://gist.github.com/LinArcX/d1d9742806fcc12624eedfcb5866a7d4

Reply:
Yes, but also including `wrapProgram $out/bin/persepolis --prefix PATH : "${lib.makeBinPath [aria libnotify ]}"`, assuming `persepolis` is the only binary.

Reply:
Done!

Reply:
Awesome!